src/main/java/org/candlepin/resource/EnvironmentResource.java

```diff
@@ -191,22 +192,24 @@ public EnvironmentDTO getEnvironment(@Verify(Environment.class) String envId) {
     }
 
     @Override
-    public void deleteEnvironment(@Verify(Environment.class) String envId) {
+    public void deleteEnvironment(@Verify(Environment.class) String envId, Boolean deleteConsumers) {
         Environment environment = envCurator.get(envId);
         if (environment == null) {
             throw new NotFoundException(i18n.tr("No such environment: {0}", envId));
         }
 
         List<Consumer> consumers = this.envCurator.getEnvironmentConsumers(environment);
-        List<Consumer> consumersToDelete = consumers.stream()
-            .filter(consumer -> consumer.getEnvironmentIds().size() == 1)
-            .toList();
-        deleteConsumers(environment, consumersToDelete);
 
-        List<Consumer> consumersToKeep = consumers.stream()
-            .filter(consumer -> consumer.getEnvironmentIds().size() > 1)
-            .toList();
-        removeConsumersFromEnvironment(environment, consumersToKeep);
+        if (Boolean.TRUE.equals(deleteConsumers)) {
+            Map<Boolean, List<Consumer>> partitionedConsumers = consumers.stream()
```

Comment:
With this approach we should save time because we won't have to go through the list twice but only once.